packages/slate-editor/src/extensions/placeholders/PlaceholdersExtension.tsx

```diff
@@ -211,6 +211,20 @@ export function PlaceholdersExtension({
                 );
             }
             if (withEmbedPlaceholders && isPlaceholderNode(element, PlaceholderNode.Type.EMBED)) {
+                if (element.provider === PlaceholderNode.Provider.INSTAGRAM) {
+                    return (
+                        <EmbedPlaceholderElement
+                            attributes={attributes}
+                            element={element}
+                            fetchOembed={withEmbedPlaceholders.fetchOembed}
+                            format={format}
+                            removable={removable}
+                        >
+                            {children}
+                        </EmbedPlaceholderElement>
+                    );
+                }
```

Comment:
I feel like having these extra IFs here makes it too crowded.

How about moving the conditions into the specific placeholder components?
E.g. the Instagram placeholder support will be implemented in the `EmbedPlaceholderElement`. Youtube will sit in the `VideoPlaceholderElement`. And so on.

What would you say?